lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -128,7 +128,16 @@ def generate_refresh_token
     end
 
     def generate_token
-      self.token = UniqueToken.generate
+      generator = Doorkeeper.configuration.access_token_generator.constantize
+
+      if generator.method(:generate).arity == 1
+        self.token = generator.generate(resource_owner_id)
+      else
+        self.token = generator.generate
+      end
+    rescue ArgumentError, TypeError, NoMethodError, NameError
+      raise Errors::UnableToGenerateToken,
+      "#{generator} does not respond to `.generate`."
```

Comment:
I think it expects two spaces of indentation, to make it clear that it's part of the previous statement, and not a new one.